totp/src/main/java/com/okta/totp/otp_display/OtpDisplayScreen.kt

```diff
@@ -0,0 +1,195 @@
+                    contentDescription = stringResource(id = R.string.otp_screen_add_button_description)
+                )
+            }
+        }
+    ) { padding ->
+        OtpScreenList(
+            otpEntryList = otpEntryList,
+            modifier = Modifier.padding(padding),
+        )
+    }
+}
+
+@Composable
+fun OtpScreenList(otpEntryList: List<OtpEntry>, modifier: Modifier = Modifier) {
+    LazyColumn(
+        modifier = modifier.padding(16.dp),
+        verticalArrangement = Arrangement.spacedBy(16.dp),
+    ) {
+        items(otpEntryList) { otpData ->
+            OtpCode(otpEntry = otpData)
+        }
+    }
+}
+
+@Composable
+fun OtpCode(otpEntry: OtpEntry, modifier: Modifier = Modifier) {
+    Row(modifier = modifier) {
+        Column {
+            Text(
+                otpEntry.otpCode,
+                fontWeight = FontWeight.Bold,
+                color = Color.Blue,
+                fontSize = 16.sp
+            )
+            Text(stringResource(id = R.string.account_label, otpEntry.account))
+            otpEntry.issuer?.let {
+                Text(stringResource(id = R.string.issuer_label, otpEntry.issuer))
+            }
+        }
+        Spacer(modifier = Modifier.weight(1f))
+        Button(
+            modifier = Modifier
+                .align(Alignment.CenterVertically)
+                .padding(16.dp),
+            onClick = { otpEntry.delete() },
+        ) {
+            Icon(
+                imageVector = Icons.Filled.Delete,
+                contentDescription = stringResource(id = R.string.delete_button_description),
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+private fun PreviewOtpScreenList() {
+    val context = LocalContext.current
+    OtpScreenList(
+        listOf(
+            OtpEntry(
+                "123456789",
+                "accoun1t@gmail.com",
+                "issuer1",
+                onDeleteOtpEntry = {
+                    Toast.makeText(
+                        context,
+                        "Deleting entry 1",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ),
+            OtpEntry(
+                "321412341",
+                "account2@gmail.com",
+                "issuer2",
+                onDeleteOtpEntry = {
+                    Toast.makeText(
+                        context,
+                        "Deleting entry 2",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ),
+            OtpEntry(
+                "653156123",
+                "account3@gmail.com",
+                "issuer3",
+                onDeleteOtpEntry = {
+                    Toast.makeText(
+                        context,
+                        "Deleting entry 3",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+            ),
+        ),
+    )
+}
+
+data class OtpEntry(
```

Comment:
Probably makes sense to remove the data modifier is we're doing custom equals/hashcode.